File: Clase 12-02-2022/1443 - DPA.cpp
#include <bits/stdc++.h>
using namespace std;
 
int sumaDeDivisores(int n) {
    int suma = 0;
    for(int i = 1; i < n; i++)
        if(n % i == 0)
            suma += i; //suma = suma + i;
    return suma;
}
 
int main(){  
    int x;
    while(cin >> x){
        int res = sumaDeDivisores(x);
        if(res == x)
            cout << "Perfecto" << endl;
        else if(res < x)
            cout << "Deficiente" << "\n";
        else
            cout << "Abundante\n";
    }
    return 0;
}

File: Clase 12-02-2022/1443 - DPA.java
import java.util.*;
 
public class Main{
    private static Scanner in;
 
    public static int sumaDeDivisores(int n) {
        int suma = 0;
        for(int i = 1; i < n; i++)
            if(n % i == 0)
                suma += i; //suma = suma + i;
        return suma;
    }
     
    public static void main(String[] args){
        in = new Scanner(System.in);
        while(in.hasNext()) {
            int x = in.nextInt();
            int res = sumaDeDivisores(x);
            if(res == x)
                System.out.println("Perfecto");
            else if(res < x)
                System.out.println("Deficiente");
            else
                System.out.println("Abundante");
        }
    }
}
 